core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -543,6 +543,53 @@ class DAGSchedulerSuite extends FunSuiteLike  with BeforeAndAfter with LocalSpar
     assertDataStructuresEmpty
   }
 
+  test("Make sure mapStage.pendingtasks is set() " +
+    "while MapStage.isAvailable is true while stage was retry ") {
```

Comment:
can you provide a more concise (less than or equal to 5 words) description for this tests? It would be good to have a more concise title and the description of the PR as well.